identity-verification/module_2/2.4 Improving Face Match Accuracy Example.ipynb

```diff
@@ -0,0 +1,521 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0d374757-07d4-4c12-93b1-b7885a074599",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Step 0 - Load libraries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3187d076-6619-4bbe-9891-4b248f298685",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install -qU opencv-python-headless\n",
+    "import boto3, os, io, glob, cv2\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline \n",
+    "client=boto3.client('rekognition')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2f289628-f8ad-487a-b27f-088470f785ca",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Step 1 - List exisiting collections "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e0da008e-34cf-4eec-ab1c-fd945c6ca512",
+   "metadata": {},
+   "source": [
+    "Before we create new collections, let's have a look if there are any exisiting collections in our account."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "eb26fb06-05e9-4908-8e5d-c3b9a586f14b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def list_collections():\n",
+    "\n",
+    "    max_results=10\n",
+    "    \n",
+    "    print('Displaying collections...')\n",
+    "    response=client.list_collections(MaxResults=max_results)\n",
+    "    collection_count=0\n",
+    "    done=False\n",
+    "    \n",
+    "    while not done:\n",
+    "        collections=response['CollectionIds']\n",
+    "\n",
+    "        for collection in collections:\n",
+    "            print (collection)\n",
+    "            collection_count+=1\n",
+    "        if 'NextToken' in response:\n",
+    "            nextToken=response['NextToken']\n",
+    "            response=client.list_collections(NextToken=nextToken,MaxResults=max_results)\n",
+    "            \n",
+    "        else:\n",
+    "            done=True\n",
+    "\n",
+    "    return collection_count   \n",
+    "\n",
+    "collection_count=list_collections()\n",
+    "\n",
+    "print(\"There are: {} collections in your account \".format(collection_count))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5dea9be7-ce31-44aa-b1f2-c7da6d13867d",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Step 2 - Create new collections"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7ebfdd26-dcd9-49db-a7db-d426377b4d85",
+   "metadata": {},
+   "source": [
+    "In this section we will create two collections, in order to compare results when there is a single face in the collection versus multiple faces profiles."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4e1618b0-9ada-44b3-a1eb-4dca4b781572",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "collection_A='collectionA' # name your collection A\n",
+    "collection_B='collectionB' # name your collection B\n",
```

Comment:
```suggestion
    "collection_A='' # name your collection A\n",
    "collection_B='' # name your collection B\n",
```

I think we tend to prefer to leave blank defaults, because if a customer mistakenly already has `collectionA` in their account and click `play` in Jupyter without paying good attention, they may have a failure here, but by hitting step 6 (delete collection), they may actually delete an existing collection losing data. If we instead default to blank, both this and step 6 should have no data loss risk. Do you agree?

Reply:
Makes sense!